kegg_pathways_completeness/bin/plot_modules_graphs.py

```diff
@@ -105,17 +165,39 @@ def generate_plot_for_completeness(self):
                 graph = self.graphs[name]
                 logging.info(f'Plotting {name}')
                 presented_ko = self.modules_completeness[name].split(',')
-                dot = self.create_graph(name, presented=presented_ko, graph=graph,
+                if self.use_pydot:
+                    logging.info('Using pydot')
+                    dot = self.create_graph_dot(name, presented=presented_ko, graph=graph,
                                         pathways_schema=self.modules_definitions[name])
-                dot.render(directory=self.outdir, filename=name, format='png')
+                    # create .dot file
+                    with open(os.path.join(self.outdir_dot, f"{name}.dot"), "w") as f:
+                        f.write(dot.to_string())
+                    # create .png file
+                    dot.write_png(os.path.join(self.outdir_png, f'{name}.png'))
+                else:
+                    logging.info('Using graphviz')
+                    dot = self.create_graph(name, presented=presented_ko, graph=graph,
+                                            pathways_schema=self.modules_definitions[name])
+                    dot.render(directory=self.outdir, filename=name, format='png')
         else:
```

Comment:
In my opinion, the generate_plot_for_completeness implies that its for plotting modules with completeness information. Therefore, if there are "things" that do not have completeness information, then they should not use this function. So basically, instead of this function having a big if else block, there should be one function for completeness and another one for lists without completeness. 
However, this is just a matter of opinion I think. So feel free to ignore.

Reply:
That is brilliant catch! I will modify names and functions. Thank you!